frontstage/common/api_call.py

```diff
@@ -0,0 +1,39 @@
+import logging
+
+import requests
+from requests.exceptions import ConnectionError, ConnectTimeout
+from structlog import wrap_logger
+
+from frontstage import app
+from frontstage.exceptions.exceptions import FailedRequest, InvalidRequestMethod
+
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+def api_call(method, endpoint, parameters=None, json=None, headers=None):
+    url = app.config['RAS_FRONTSTAGE_API_SERVICE'] + endpoint
+    if parameters:
+        url = '{}?'.format(url)
+        for key, value in parameters.items():
+            parameter_string = '{}={}&'.format(key, value)
+            url = url + parameter_string
+    try:
+        if method == 'GET':
+            response = requests.get(url, headers=headers)
+        elif method == 'POST':
+            response = requests.post(url, headers=headers, json=json)
+        elif method == 'PUT':
+            response = requests.put(url, headers=headers, json=json)
+        else:
+            logger.error('Invalid request method', method=str(method), url=url)
+            raise InvalidRequestMethod(method, url)
+    except ConnectTimeout as e:
+        logger.error('Connection to remote server timed out', method=method, url=url, exception=str(e))
+        raise FailedRequest(method, url, e)
```

Comment:
Is there much value in wrapping and raising another type of exception?

Reply:
I think we'll want to catch any failed calls to the API explicitly so we can flag it up somewhere (Splunk etc). All the calls frontstage makes will be going through it so it'll be important to know when there's failures there.
I could make another `APIFailure` exception maybe which could handle it?

Reply:
I could add an error handler for it which returns the general error page and logs a message we can look for in splunk?

Reply:
I was thinking we'd could just catch ConnectTimeout and ConnectionError in the error handler